src/main/java/frc/robot/subsystems/Mechanism.java

```diff
@@ -0,0 +1,374 @@
+        this.m_sourceMotor.setOpenLoopRampRate(0.05);
+        this.m_ampMotor.setOpenLoopRampRate(0.05);
+
+        this.m_bottomBeltMotor.enableVoltageCompensation(12);
+        this.m_topBeltMotor.enableVoltageCompensation(12);
+        this.m_sourceMotor.enableVoltageCompensation(12);
+        this.m_ampMotor.enableVoltageCompensation(12);
+
+        requestIntakeType = 0;
+    }
+
+    /**
+     * Gets the instance of the mechanism
+     * @return the instance of the mechanism
+     */
+    public static Mechanism getInstance() {
+        if (instance == null) {
+            instance = new Mechanism();
+        }
+        return instance;
+    }
+
+    /**
+     * updates the phase of the beam break sensor
+     */
+    private void updatePhase() {
+        this.m_beamBreak.updatePhase();
+        this.m_currentPhase = this.m_beamBreak.getPhase();
+    }
+
+    /**
+     * Sets the speed of the belt motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setBeltSpeed(double speed) {
+        this.m_bottomBeltMotor.setVoltage(speed);
+        this.m_topBeltMotor.setVoltage(speed);
+    }
+
+    /**
+     * Sets the speed of the source motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setSourceSpeed(double speed) {
+        this.m_sourceMotor.setVoltage(speed);
+    }
+    
+    /**
+     * Sets the speed of the amp motor
+     * @param speed the speed in volts [0 --> 12]
+     */
+    private void setAmpSpeed(double speed) {
+        this.m_ampMotor.setVoltage(speed);
+    }
+
+    /**
+     * A method to check if the phase changed
+     * @param phase the phase that is checked
+     */
+    public boolean checkState(Phase phase) {
+        return m_currentPhase == phase;
+    }
+
+    /**
+     * A function for requesting ground/source intake using LEDs
+     * @param intakeType The requested intake type (1 is ground, 2 is source, 0 is nothing)
+     */
+    public void requestIntake(int intakeType) {
+        // Set the requested intake based on the input
+        if (requestIntakeType == intakeType) { // If you request the same intake twice the lights turn off
+            requestIntakeType = 0;
+        }
+        else {
+            requestIntakeType = intakeType;
+        }
+
+        // Set the LED color to the requested intake
+        if (this.m_currentPhase != Phase.NONE) {
+            requestIntakeType = 0;
+        }
+        else if (requestIntakeType == 2) {
+            this.powerLEDs("cyan");
+        }
+        else if (requestIntakeType == 1) {
+            this.powerLEDs("purple");
+        }
+        else {
+            this.powerLEDs("off");
+        }
+    }
+
+    /**
+     * ground intake command
+     * @param speed the speed to run the ground intake at in volts [0 --> 12]
+     * @return
+     */
+    public Command groundIntake(double speed) {
+        return this.startEnd(
+            () -> {
+                this.setBeltSpeed(-speed);
```

Comment:
consider extracting these 3 method calls into one method i.e. 

`setIntake(double belt, double source, double amp, double speed);`

Reply:
As it is used many times through this class

Reply:
Not really, as you can just feed the - and + speed into it when needed. I think it is too much code that can be simplified.